libvirt/tests/src/libvirt_rng.py

```
@@ -342,6 +343,14 @@ def check_guest(session, expect_fail=False):
         rng_currt = session.cmd_output("cat %s" % rng_files[1],
                                        timeout=timeout).strip()
         logging.debug("rng avail:%s, current:%s", rng_avail, rng_currt)
+        if not rng_avail.count("virtio"):
```

this is already covered by later line 354

---

Yes, please remove this line.

---

Is this line to avoid empty for L349? If yes, could it be moved to be inside "if set_virtio_current:'?
Other LGTM.

---

> this is already covered by later line 354

I think there's a misunderstanding, l. 354 checks the `rng_currt`, ie. if the attached virtio device has been set as the one that's currently used by the kernel, whereas here we check the precondition that the attached virtio is available at all.

---

> Yes, please remove this line.

This line is needed because as pointed out by @yanqzhan it's there to make sure there's a virtio device available when we want to update `rng_currt` to it. The test should fail early if there wasn't a virtio device `rng_avail`; this also should guarantee that the test failure error message is correct (instead of a IndexNotFound error where the code accesses the first available virtio device via indexing over the list comprehension `[0]`).

---

> Is this line to avoid empty for L349? If yes, could it be moved to be inside "if set_virtio_current:'? Other LGTM.

Yes :) This should avoid an "IndexError: list index out of range"
Theoretically, it could be moved inside of the block that you mention. However, I prefer not to because when the function `check_guest` is called we assume that a virtio device is available, this is not new in this PR but also in existing code. Let's suppose `set_virtio_current == False` but there's no device available, then the tests would fail in the later line `if not rng_currt.count("virtio") or rng_currt not in rng_avail` but the check is more specific as it not only expects virtio to be available but additionally to be the current device selected by the kernel. I'll update the error messages to make the difference more clear. Please, let me know what you think.